pkg/workloads/console/events/gcp_pubsub.go

```diff
@@ -0,0 +1,69 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/pubsub"
+)
+
+type ErrorNoTopic struct {
+	Topic string
+}
+
+func (e ErrorNoTopic) Error() string {
+	return fmt.Sprintf("No topic found with id: %s", e.Topic)
+}
+
+var _ error = &ErrorNoTopic{}
+
+type GooglePubSubPublisher struct {
+	client *pubsub.Client
+	topic  *pubsub.Topic
+}
+
+var _ Publisher = &GooglePubSubPublisher{}
+
+func NewGooglePubSubPublisher(ctx context.Context, projectName string, topicName string) (*GooglePubSubPublisher, error) {
+	client, err := pubsub.NewClient(ctx, projectName)
+	if err != nil {
+		return nil, err
```

Comment:
Not sure what kind of errors we'd get back from the library, so some error wrapping might not go amiss here (and below in the same file) to keep things understandable when it goes wrong.